lassen/sim.py

```diff
@@ -27,168 +27,191 @@
 #   C (carry generated)
 #   V (overflow generated)
 #
-def gen_alu(family: TypeFamily, datawidth):
-    Bit = family.Bit
-    Data = family.BitVector[datawidth]
+class ALU(Peak):
+    def __init__(self,family: TypeFamily, datawidth=16):
+        super(ALU,self).__init__(family,datawidth)
+        #self.Bit = family.Bit
+        #self.Data = family.BitVector[datawidth]
+        #self.Signed = family.Signed
+        #self.BitVector = family.BitVector
```

Comment:
This is a pretty significant departure from the old syntax. Basically, magma requires all types to be known at compile time (looking at a peak definition), so having the wrapper `gen_alu` function allows us to stage the definition of the types. In this new model, it seems like the family is defined upon instantiation, which significantly complicates the magma logic (we'll need to follow this instance chain and define the types as we go). Is this the plan to move forward? I would vote that we revert to the function syntax at least until after the EOQ.